twitter/twitter.php

```diff
@@ -1063,34 +1071,52 @@ function twitter_get_relation($uid, $target, $contact = [])
 	return $relation;
 }
 
-function twitter_fetch_contact($uid, $data, $create_user)
+function twitter_user_to_contact($data)
 {
 	if (empty($data->id_str)) {
 		return -1;
 	}
 
 	$avatar = twitter_fix_avatar($data->profile_image_url_https);
-	$baseurl = "https://twitter.com";
-	$url = $baseurl . "/" . $data->screen_name;
-	$addr = $data->screen_name . "@twitter.com";
+	$baseurl = 'https://twitter.com';
+	$url = $baseurl . '/' . $data->screen_name;
+	$addr = $data->screen_name . '@twitter.com';
+
+	$fields = [
+		'url'      => $url,
+		'network'  => Protocol::TWITTER,
+		'alias'    => 'twitter::' . $data->id_str,
+		'baseurl'  => $baseurl,
+		'name'     => $data->name,
+		'nick'     => $data->screen_name,
+		'addr'     => $addr,
+		'location' => $data->location,
+		'about'    => $data->description,
+		'photo'    => $avatar,
```

Comment:
In the contact table the field "avatar" is meant to store the real path to the avatar picture. In "photo" there is the cropped and cached version.

Reply:
Got it.